### PR TITLE
Updating the href on save complete, and awaiting the save of the activity usage

### PR DESCRIPTION
**Pull request by quinnbudan**
## Relevant Rally Links 

- [DE43265](https://rally1.rallydev.com/#/289692574792d/dashboard?detail=%2Fdefect%2F600684803540&fdp=true?fdp=true): Lessons FACE Weblinks >  Saving a new weblink sometimes causes a 500 error
- [DE43284](https://rally1.rallydev.com/#/289692574792d/dashboard?detail=%2Fdefect%2F600899688384&fdp=true?fdp=true): Lessons FACE Weblinks > Visibility status and dates are not saved when creating a new weblink with working copy enabled

## Description

We want to adjust the save flow for content FACE pages so that on saving a new activity we re-fetch the activity usage if we're not navigating away from the FACE page. 

## Goal/Solution

The solution was to stop the editor from stopping the save complete event propagation and updating the `d2l-activity-content-editor` to handle the in place save complete event and set `this.href` to `this.postCommitHref`, which should have been set when the `d2l-content-working-copy-committed` event was fired. Another change was to add an `await` for the activity usage save so that the save complete event didn't fire before the usage was saved. This was necessary for us to ensure that the usage was updated properly before we re-fetched it. Finally, in order to get these fixes working for content files as well, we had to update the `saveOrder` to 500 so that the content file entity was saved prior to the activity usage.

The impact on other FACE pages should be low, since `d2l-activity-content-editor` seems to be the only editor that ever handled the save complete event, and waiting for the usage to finish saving shouldn't cause any issues either.

## Video/Screenshots

N/A

## Related PRs

- https://github.com/Brightspace/lms/pull/10084 This is the draft PR with most of the backend work we need
- https://github.com/Brightspace/lms/pull/9920 The rest of the backend work is in this PR, and the PR above points to the `compare` branch of this PR

## Remaining Work

N/A

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-content-editor/d2l-activity-content-editor.js [hunk 1]
```diff
@@ -51,18 +51,26 @@ class ContentEditor extends LocalizeActivityEditorMixin(RtlMixin(ActivityEditorM
...
+		this.preCommitHref = null;
+		this.postCommitHref = null;
+		this.onSaveComplete = null;
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
-		this.addEventListener('d2l-activity-editor-save-complete', this._redirectOnSaveComplete);
+		// storing onSaveComplete handler in property in order to have 'this' work as expected,
+		// see https://lit.dev/docs/components/events/#understanding-this-in-event-listeners and
```
Comment (ianlaidlaw) [inline components/d2l-activity-editor/d2l-activity-content-editor/d2l-activity-content-editor.js:62]: > and
did you mean to include another link here?

Comment (quinnbudan) [reply components/d2l-activity-editor/d2l-activity-content-editor/d2l-activity-content-editor.js:62]: oh yes I did, thank you! I'll add it now

**Review by CodeBaboon — APPROVED** [review]
LGTM

**Review by brissons — APPROVED** [review]
lgtm!

**Review by DiljotSG — APPROVED** [review]

**Review by ChrisKraljevicD2L — APPROVED** [review]

**Pull request merged**